README.md
![Logo](banner.jpg)

<p align="center">
    <a href="https://github.com/TheAnkurPanchani/flutter_swiper/pulls">
        <img src="https://img.shields.io/badge/PRs-Welcome-brightgreen.svg" alt="PRs Welcome" />
    </a>
    <a href="https://pub.dev/packages/card_swiper">
        <img src="https://img.shields.io/pub/v/card_swiper.svg" alt="pub package" />
    </a>
</p>

# card_swiper

Swiper/Carousel for flutter, with multiple layouts, infinite loop. Compatible with Android & iOS.

# :sparkles::sparkles: New Features:Layout for pagination.

We are using this project [flutter_page_indicator](https://github.com/best-flutter/flutter_page_indicator) now .

# :sparkles::sparkles: New Features:PageTransformer

Finally, we have `PageTransformer` like android, just set a `transformer` to `Swiper`,
it returns a widget that has been transformed. For now, only support for layout `DEFAULT`.
Thanks to @FlutterRocks ,you've done great job 👏.

We are using this project [transformer_page_view](https://github.com/best-flutter/transformer_page_view) now .

# :sparkles::sparkles: New Features:Layout

![](https://github.com/jzoom/images/raw/master/layout1.gif)

![](https://github.com/jzoom/images/raw/master/layout2.gif)

![](https://github.com/jzoom/images/raw/master/layout3.gif)

[See More](#build-in-layouts)

# Showcases

![Horizontal](https://github.com/jzoom/flutter_swiper/raw/master/example/res/1.gif)

![Vertical](https://github.com/jzoom/flutter_swiper/raw/master/example/res/2.gif)

![Custom Pagination](https://github.com/jzoom/flutter_swiper/raw/master/example/res/3.gif)

![Custom Pagination](https://github.com/jzoom/flutter_swiper/raw/master/example/res/4.gif)

![Phone](https://github.com/jzoom/flutter_swiper/raw/master/example/res/5.gif)

![Example](https://github.com/jzoom/images/raw/master/swiper-example.gif)

[See More](#codes)

## Changelogs

> see:[CHANGELOG.md](https://github.com/TheAnkurPanchani/flutter_swiper/blob/master/CHANGELOG.md)

## Getting Started

- [Installation](#installation)
- [Basic Usage](#basic-usage)
- [Constructor](#constructor)
  - [Basic](#basic)
  - [Pagination](#pagination)
  - [Control buttons](#control-buttons)
  - [Controller](#controller)
  - [Autoplay](#autoplay)
- [Build in layouts](#build-in-layouts)
- [Codes](#codes)

### Installation

Add

```bash
card_swiper : ^2.0.1
```

to your `pubspec.yaml`, and run

```bash
flutter packages get
```

in your project's root directory.

OR

run

```bash
flutter pub add card_swiper
```

in your project's root directory.

### Basic Usage

Create a new project with command

```
flutter create myapp
```

Edit lib/main.dart like this:

```dart

import 'package:flutter/material.dart';

import 'package:card_swiper/card_swiper.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
    body: Swiper(
        itemBuilder: (BuildContext context,int index){
          return Image.network("https://via.placeholder.com/350x150",fit: BoxFit.fill,);
        },
        itemCount: 3,
        pagination: SwiperPagination(),
        control: SwiperControl(),
      ),
    );
  }
}

```

### Constructor

#### Basic

| Parameter       |            Default             | Description                                                                                                                                                              |
| :-------------- | :----------------------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| scrollDirection |        Axis.horizontal         | If `Axis.horizontal`, the scroll view's children are arranged horizontally in a row instead of vertically in a column.                                                   |
| axisDirection   |       AxisDirection.left       | If `AxisDirection.right`, the scroll view's children are arranged right side in a row instead of left.<br>**Note:** Currently supported for horizontal stack layout only |
| loop            |              true              | Set to `false` to disable continuous loop mode.                                                                                                                          |
| index           |               0                | Index number of initial slide.                                                                                                                                           |
| autoplay        |             false              | Set to `true` enable auto play mode.                                                                                                                                     |
| onIndexChanged  | void onIndexChanged(int index) | Called with the new index when the user swiped or autoplay                                                                                                               |
| onTap           |     void onTap(int index)      | Called when user tap ui.                                                                                                                                                 |
| duration        |             300.0              | The milliscends of every transaction animation costs                                                                                                                     |
| pagination      |              null              | set `SwiperPagination()` to show default pagination                                                                                                                      |
| control         |              null              | set `SwiperControl()` to show default control buttons                                                                                                                    |

#### Pagination

The pagination extends from `SwiperPlugin`,the `SwiperPlugin` provides extra ui for `Swiper`.Set `SwiperPagination()` to show default pagination.

| Parameter |          Default           | Description                                                                                                                               |
| :-------- | :------------------------: | :---------------------------------------------------------------------------------------------------------------------------------------- |
| alignment |   Alignment.bottomCenter   | Change this value if you what to put pagination in other place                                                                            |
| margin    | const EdgeInsets.all(10.0) | The distance between inner side of the parent container.                                                                                  |
| builder   |   SwiperPagination.dots    | There are three default styles `SwiperPagination.dots`, `SwiperPagination.fraction` and `SwiperPagination.rect`, these can be customized. |

If you'd like to customize your own pagination, you can do like this:

```dart
Swiper(
    ...,
    pagination: SwiperCustomPagination(
        builder:(BuildContext context, SwiperPluginConfig config){
            return YourOwnPagination();
        }
    )
);

```

##### Implemented pagination

###### DotSwiperPaginationBuilder

| Parameter   | Default                                   | Description              | Required |
| :---------- | :---------------------------------------- | :----------------------- | :------: |
| activeColor | Theme.of(context).primaryColor            | Active bullet color      |  false   |
| color       | Theme.of(context).scaffoldBackgroundColor | Bullet color             |  false   |
| activeSize  | 10.0                                      | Active bullet size       |  false   |
| size        | 10.0                                      | Bullet size              |  false   |
| space       | 3.0                                       | Distance between bullets |  false   |
| key         | -                                         | key                      |  false   |

###### FractionPaginationBuilder

| Parameter      | Default                                   | Description       | Required |
| :------------- | :---------------------------------------- | :---------------- | :------: |
| activeColor    | Theme.of(context).primaryColor            | Active font color |  false   |
| color          | Theme.of(context).scaffoldBackgroundColor | font color        |  false   |
| activeFontSize | 35.0                                      | Active font size  |  false   |
| fontSize       | 20.0                                      | Font size         |  false   |
| key            | -                                         | key               |  false   |

###### RectSwiperPaginationBuilder

| Parameter   | Default                                   | Description              | Required |
| :---------- | :---------------------------------------- | :----------------------- | :------: |
| activeColor | Theme.of(context).primaryColor            | Active bullet color      |  false   |
| color       | Theme.of(context).scaffoldBackgroundColor | Bullet color             |  false   |
| activeSize  | 10.0                                      | Active bullet size       |  false   |
| size        | 10.0                                      | Bullet size              |  false   |
| space       | 3.0                                       | Distance between bullets |  false   |
| key         | -                                         | key                      |  false   |

#### Control buttons

The control also extends from `SwiperPlugin`,set `SwiperControl()` to show default control buttons.

| Parameter    |             Default             | Description                                        |
| :----------- | :-----------------------------: | :------------------------------------------------- |
| iconPrevious |      Icons.arrow_back_ios       | The icon data to display `previous` control button |
| iconNext     |     Icons.arrow_forward_ios     | The icon data to display `next`.                   |
| color        | Theme.of(context).primaryColor  | Control button color                               |
| disableColor | Theme.of(context).disabledColor | Disabled control button color                      |
| size         |              30.0               | Control button size                                |
| padding      |    const EdgeInsets.all(5.0)    | Control button padding                             |

#### Controller

The `Controller` is used to control the `index` of the Swiper, start or stop autoplay.You can create a controller by `SwiperController()` and save the instance by futher usage.

| Method                                       | Description                                         |
| :------------------------------------------- | :-------------------------------------------------- |
| void move(int index, {bool animation: true}) | Move to the spicified `index`,with animation or not |
| void next({bool animation: true})            | Move to next                                        |
| void previous({bool animation: true})        | Move to previous                                    |
| void startAutoplay()                         | Start autoplay                                      |
| void stopAutoplay()                          | Stop autoplay                                       |

#### Autoplay

| Parameter                    | Default | Description                                          |
| :--------------------------- | :-----: | :--------------------------------------------------- |
| autoplayDelay                |  3000   | Autoplay delay milliseconds.                         |
| autoplayDisableOnInteraction |  true   | If set true, `autoplay` is disabled when use swipes. |

## Build in layouts

![](https://github.com/jzoom/images/raw/master/layout1.gif)

```dart
Swiper(
  itemBuilder: (BuildContext context, int index) {
    return Image.network(
      "https://via.placeholder.com/288x188",
      fit: BoxFit.fill,
    );
  },
  itemCount: 10,
  viewportFraction: 0.8,
  scale: 0.9,
)

```

![](https://github.com/jzoom/images/raw/master/layout2.gif)

```dart
Swiper(
  itemBuilder: (BuildContext context, int index) {
    return Image.network(
      "https://via.placeholder.com/288x188",
      fit: BoxFit.fill,
    );
  },
  itemCount: 10,
  itemWidth: 300.0,
  layout: SwiperLayout.STACK,
)
```

![](https://github.com/jzoom/images/raw/master/layout3.gif)

```dart
Swiper(
    itemBuilder: (BuildContext context, int index) {
      return Image.network(
        "https://via.placeholder.com/288x188",
        fit: BoxFit.fill,
      );
    },
    itemCount: 10,
    itemWidth: 300.0,
    itemHeight: 400.0,
    layout: SwiperLayout.TINDER,
 )
```

![](https://github.com/jzoom/images/raw/master/layout4.gif)

Very easy to create you own custom animation:

```dart

Swiper(
  layout: SwiperLayout.CUSTOM,
  customLayoutOption: CustomLayoutOption(
    startIndex: -1,
    stateCount: 3
  )..addRotate([
    -45.0/180,
    0.0,
    45.0/180
  ])..addTranslate([
    Offset(-370.0, -40.0),
    Offset(0.0, 0.0),
    Offset(370.0, -40.0)
  ]),
  itemWidth: 300.0,
  itemHeight: 200.0,
  itemBuilder: (context, index) {
    return Container(
      color: Colors.grey,
      child: Center(
        child: Text("$index"),
      ),
    );
  },
  itemCount: 10,
)

```

The `CustomLayoutOption` is designed to describe animations.
It is very easy to specify every state of items in Swiper.

```dart
CustomLayoutOption(
  // Which index is the first item of array below
  startIndex: -1,
  // array length
  stateCount: 3
)..addRotate([
  // rotation of every item
  -45.0/180,
  0.0,
  45.0/180
])..addTranslate([
  // offset of every item
  Offset(-370.0, -40.0),
  Offset(0.0, 0.0),
  Offset(370.0, -40.0)
]);

```

## Codes

![Example](https://github.com/jzoom/images/raw/master/swiper-example.gif)

```dart
ConstrainedBox(
  child: Swiper(
    outer:false,
    itemBuilder: (c, i) {
      return Wrap(
        runSpacing: 6.0,
        children: [0,1,2,3,4,5,6,7,8,9].map((i){
          return SizedBox(
            width: MediaQuery.of(context).size.width/5,
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: <Widget>[
                SizedBox(
                  child: Container(
                    child: Image.network("https://fuss10.elemecdn.com/c/db/d20d49e5029281b9b73db1c5ec6f9jpeg.jpeg%3FimageMogr/format/webp/thumbnail/!90x90r/gravity/Center/crop/90x90"),
                  ),
                  height: MediaQuery.of(context).size.width * 0.12,
                  width: MediaQuery.of(context).size.width * 0.12,
                ),
                Padding(padding: EdgeInsets.only(top:6.0),child: Text("$i"),)
              ],
            ),
          );
        }).toList(),
      );
    },
    pagination: SwiperPagination(
      margin: EdgeInsets.all(5.0)
    ),
    itemCount: 10,
  ),
    constraints:BoxConstraints.loose(Size(screenWidth, 170.0))
),

```

# FAQs

## How to remove default controlls?

- Do not pass any value in `control` parameter of constructor

## How to remove default pagination?

- Do not pass any value in `pagination` parameter of constructor

---

You can find all custom options here:

> https://github.com/TheAnkurPanchani/flutter_swiper/blob/master/example/lib/src/ExampleCustom.dart


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added

### Changed

### Deprecated

### Removed

### Fixed

### Security

## [2.0.4] - 2022-06-05

### Fixed

- [PR#46](https://github.com/TheAnkurPanchani/card_swiper/pull/46): `dart fix --apply` to support Flutter SDK 3.0
- [Issue#47](https://github.com/TheAnkurPanchani/card_swiper/issues/47): cant work flutter 3.0
- [Issue#48](https://github.com/TheAnkurPanchani/card_swiper/issues/48): Not supported for Flutter 3.0
- [Issue#49](https://github.com/TheAnkurPanchani/card_swiper/issues/49): 建议升级一下3.0
- [Issue#50](https://github.com/TheAnkurPanchani/card_swiper/issues/50): not supporting v3 flutter

## [2.0.3] - 2022-03-13

### Added

- [PR#29](https://github.com/TheAnkurPanchani/card_swiper/pull/29): Add allowImplicitScrolling property

### Fixed

- [PR#28](https://github.com/TheAnkurPanchani/card_swiper/pull/28): make page indicators respect the initial page index

## [2.0.2] - 2022-03-13

### Fixed

- [Issue#12](https://github.com/TheAnkurPanchani/card_swiper/issues/12): Use default pagination only if value is not provided
- [Issue#15](https://github.com/TheAnkurPanchani/card_swiper/issues/15): Put item count check before building animation
- [Issue#24](https://github.com/TheAnkurPanchani/card_swiper/issues/24): Put mounted check on post-frame callbacks
- [Issue#30](https://github.com/TheAnkurPanchani/card_swiper/issues/30): Fix order for right direction of stack layout
- [Issue#31](https://github.com/TheAnkurPanchani/card_swiper/issues/31): Update README for custom layout options
- [Issue#32](https://github.com/TheAnkurPanchani/card_swiper/issues/32): Update README to mention how to not get default controls and pagination

## [2.0.1] - 2022-01-11

### Added

- [Issue#3](https://github.com/TheAnkurPanchani/card_swiper/issues/3): Support `AxisDirection` left and right for Stack layout

### Fixed

- Fix pan event index update for custom layout

## [2.0.0] - 2022-01-09

### Changed

- [PR#17](https://github.com/TheAnkurPanchani/card_swiper/pull/17): Migrate from `pedantic` to `flutter_lints_plus` and fix problems related to lint

## [1.0.4] - 2021-08-06

### Fixed

- Initial index should set from controller is provided in parameter
- Update readme

## [1.0.3] - 2021-08-06

### Fixed

- Initial index should set from controller is provided in parameter

## [1.0.2] - 2017-06-20

### Fixed

- README text and references
- Library name
- Example reference
- Format according to dartfmt

## [1.0.1] - 2017-06-20

### Fixed

- Format according to dartfmt

## [1.0.0] - 2017-06-20

- First release after fork


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_swiper/flutter_swiper.dart';

import 'src/ExampleCustom.dart';
import 'src/ExampleSwiperInScrollView.dart';
import 'src/config.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: new MyHomePage(title: 'Flutter Swiper'),
      //home: buildHome(),
      routes: {
        '/example01': (BuildContext context) => new ExampleHorizontal(),
        '/example02': (BuildContext context) => new ExampleVertical(),
        '/example03': (BuildContext context) => new ExampleFraction(),
        '/example04': (BuildContext context) => new ExampleCustomPagination(),
        '/example05': (BuildContext context) => new ExamplePhone(),
        '/example06': (BuildContext context) => new ScaffoldWidget(
            child: new ExampleSwiperInScrollView(), title: "ScrollView"),
        '/example07': (BuildContext context) => new ScaffoldWidget(
              child: new ExampleCustom(),
              title: "Custom All",
            )
      },
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => new _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<Widget> render(BuildContext context, List children) {
    return ListTile.divideTiles(
        context: context,
        tiles: children.map((dynamic data) {
          return buildListTile(context, data[0], data[1], data[2]);
        })).toList();
  }

  Widget buildListTile(
      BuildContext context, String title, String subtitle, String url) {
    return new ListTile(
      onTap: () {
        Navigator.of(context).pushNamed(url);
      },
      isThreeLine: true,
      dense: false,
      leading: null,
      title: new Text(title),
      subtitle: new Text(subtitle),
      trailing: new Icon(
        Icons.arrow_right,
        color: Colors.blueAccent,
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    // DateTime moonLanding = DateTime.parse("1969-07-20");

    return new Scaffold(
      appBar: new AppBar(
        title: Text(widget.title),
      ),
      body: new ListView(
        children: render(context, [
          ["Horizontal", "Scroll Horizontal", "/example01"],
          ["Vertical", "Scroll Vertical", "/example02"],
          ["Fraction", "Fraction style", "/example03"],
          ["Custom Pagination", "Custom Pagination", "/example04"],
          ["Phone", "Phone view", "/example05"],
          ["ScrollView ", "In a ScrollView", "/example06"],
          ["Custom", "Custom all properties", "/example07"]
        ]),
      ),
    );
  }
}

const List<String> titles = [
  "Flutter Swiper is awosome",
  "Really nice",
  "Yeap"
];

class ExampleHorizontal extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: AppBar(
          title: Text("ExampleHorizontal"),
        ),
        body: new Swiper(
          itemBuilder: (BuildContext context, int index) {
            return new Image.asset(
              images[index],
              fit: BoxFit.fill,
            );
          },

          indicatorLayout: PageIndicatorLayout.COLOR,
          autoplay: true,
          itemCount: images.length,
          pagination: new SwiperPagination(),
          control: new SwiperControl(),
        ));
  }
}

class ExampleVertical extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: AppBar(
          title: Text("ExampleVertical"),
        ),
        body: new Swiper(
          itemBuilder: (BuildContext context, int index) {
            return new Image.asset(
              images[index],
              fit: BoxFit.fill,
            );
          },
          autoplay: true,
          itemCount: images.length,
          scrollDirection: Axis.vertical,
          pagination: new SwiperPagination(alignment: Alignment.centerRight),
          control: new SwiperControl(),
        ));
  }
}

class ExampleFraction extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(
          title: Text("ExampleFraction"),
        ),
        body: new Column(
          children: <Widget>[
            Expanded(
                child: new Swiper(
              itemBuilder: (BuildContext context, int index) {
                return new Image.asset(
                  images[index],
                  fit: BoxFit.fill,
                );
              },
              autoplay: true,
              itemCount: images.length,
              pagination:
                  new SwiperPagination(builder: SwiperPagination.fraction),
              control: new SwiperControl(),
            )),
            Expanded(
                child: new Swiper(
              itemBuilder: (BuildContext context, int index) {
                return new Image.asset(
                  images[index],
                  fit: BoxFit.fill,
                );
              },
              autoplay: true,
              itemCount: images.length,
              scrollDirection: Axis.vertical,
              pagination: new SwiperPagination(
                  alignment: Alignment.centerRight,
                  builder: SwiperPagination.fraction),
            ))
          ],
        ));
  }
}

class ExampleCustomPagination extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(
          title: new Text("Custom Pagination"),
        ),
        body: new Column(
          children: <Widget>[
            new Expanded(
              child: new Swiper(
                itemBuilder: (BuildContext context, int index) {
                  return new Image.asset(
                    images[index],
                    fit: BoxFit.fill,
                  );
                },
                autoplay: true,
                itemCount: images.length,
                pagination: new SwiperPagination(
                    margin: new EdgeInsets.all(0.0),
                    builder: new SwiperCustomPagination(builder:
                        (BuildContext context, SwiperPluginConfig config) {
                      return new ConstrainedBox(
                        child: new Container(
                            color: Colors.white,
                            child: new Text(
                              "${titles[config.activeIndex]} ${config.activeIndex + 1}/${config.itemCount}",
                              style: new TextStyle(fontSize: 20.0),
                            )),
                        constraints: new BoxConstraints.expand(height: 50.0),
                      );
                    })),
                control: new SwiperControl(),
              ),
            ),
            new Expanded(
              child: new Swiper(
                itemBuilder: (BuildContext context, int index) {
                  return new Image.asset(
                    images[index],
                    fit: BoxFit.fill,
                  );
                },
                autoplay: true,
                itemCount: images.length,
                pagination: new SwiperPagination(
                    margin: new EdgeInsets.all(0.0),
                    builder: new SwiperCustomPagination(builder:
                        (BuildContext context, SwiperPluginConfig config) {
                      return new ConstrainedBox(
                        child: new Row(
                          children: <Widget>[
                            new Text(
                              "${titles[config.activeIndex]} ${config.activeIndex + 1}/${config.itemCount}",
                              style: TextStyle(fontSize: 20.0),
                            ),
                            new Expanded(
                              child: new Align(
                                alignment: Alignment.centerRight,
                                child: new DotSwiperPaginationBuilder(
                                        color: Colors.black12,
                                        activeColor: Colors.black,
                                        size: 10.0,
                                        activeSize: 20.0)
                                    .build(context, config),
                              ),
                            )
                          ],
                        ),
                        constraints: new BoxConstraints.expand(height: 50.0),
                      );
                    })),
                control: new SwiperControl(color: Colors.redAccent),
              ),
            )
          ],
        ));
  }
}

class ExamplePhone extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text("Phone"),
      ),
      body: new Stack(
        children: <Widget>[
          ConstrainedBox(
            constraints: new BoxConstraints.expand(),
            child: new Image.asset(
              "images/bg.jpeg",
              fit: BoxFit.fill,
            ),
          ),
          new Swiper.children(
            autoplay: false,
            pagination: new SwiperPagination(
                margin: new EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 30.0),
                builder: new DotSwiperPaginationBuilder(
                    color: Colors.white30,
                    activeColor: Colors.white,
                    size: 20.0,
                    activeSize: 20.0)),
            children: <Widget>[
              new Image.asset(
                "images/1.png",
                fit: BoxFit.contain,
              ),
              new Image.asset(
                "images/2.png",
                fit: BoxFit.contain,
              ),
              new Image.asset("images/3.png", fit: BoxFit.contain)
            ],
          )
        ],
      ),
    );
  }
}

class ScaffoldWidget extends StatelessWidget {
  final Widget? child;
  final String title;
  final List<Widget> actions;

  ScaffoldWidget({this.child, required this.title, this.actions = const []});

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text(title),
        actions: actions,
      ),
      body: child,
    );
  }
}
